src/app/shared/featured-sale-offers/sale-offers-adidas/sale-offers-adidas.component.ts

```diff
@@ -0,0 +1,26 @@
+import { Component, computed, inject, Input } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { RouterModule } from '@angular/router';
+import { SaleOfferService } from '../../../modules/offers/sale-offers/services/sale-offer.service';
+import { SaleOfferCardHomeComponent } from '../../../modules/offers/sale-offers/sale-offer-card-home/sale-offer-card-home.component';
+
+@Component({
+  selector: 'app-sale-offers-adidas',
+  imports: [
+    CommonModule,
+    RouterModule,
+    SaleOfferCardHomeComponent
+  ],
+  templateUrl: './sale-offers-adidas.component.html',
+  styleUrls: ['./sale-offers-adidas.component.scss'],
+})
+export class SaleOffersAdidasComponent {
+  @Input() brandIdFilter: number = 4; 
```

Comment:
Używa się brandFilter = input.required<number>() a nie przestarzałego @Input().

Chat generuje stare rezultaty.
Dodatkowo to do niczego nie jest tutaj potrzebne.

Zamiast tych kilku komponentów trzeba zrobić jeden tak jak rozmawialiśmy